### PR TITLE
Feature/uf lmapping

### PR DESCRIPTION
**Pull request by no-reply**
Submitting this as a PR so there is a discussion point.

### CODE REVIEW

File: heidrun/ufl_marc.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    collection :class => DPLA::MAP::Collection, 
+               :each => record.field('marc:datafield')
+                              .match_attribute(:tag, '830'),
+               :as => :coll do
+      title coll.field('marc:subfield').match_attribute(:code, 'a')
+    end
+
+    #don't know how to do these but am keeping placeholder
+    #contributor
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb]: The contributor mapping suggests that we need only `700$a` here when `$e` is not `joint author`. Do we really want to throw away the other subfields?

File: heidrun/ufl_marc.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+               :each => record.field('marc:datafield')
+                              .match_attribute(:tag, '830'),
+               :as => :coll do
+      title coll.field('marc:subfield').match_attribute(:code, 'a')
+    end
+
+    #don't know how to do these but am keeping placeholder
+    #contributor 
+
+    #creator
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb]: The creator mapping suggests that we need only `$a` for 100, 110, 111 (and `700` when `$e` is `joint author` or `jt author`). Do we really want to throw away the other subfields?

File: heidrun/ufl_marc.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    #creator 
+
+    date :class => DPLA::MAP::TimeSpan,
+         :each => record.field('marc:datafield')
+                        .match_attribute(:tag, '260'),
+         :as => :date do
+      providedLabel date.field('marc:subfield').match_attribute(:code, 'c')
+    end
+
+    #description
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb]: from mapping document: `5XX; not 506, 538, 536, 535, 533, 510, 540`, so maybe something like:

``` ruby
description record.field('marc:datafield')
                  .match_attribute(:tag) { |tag| tag =~ /^5..$/ }
                  .reject { |df| ['506', '538', '536', '535', '533', '510', '540'].include? df.tag }
```

... but then we need away to concatenate the values of all of the subfields.

File: heidrun/ufl_marc.rb [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                       !df['marc:subfield'].match_attribute(:code, 'b').empty?) }
+            .field('marc:subfield')
+    
+    #genre 
+
+    #identifier 
+
+    #language 
+
+    #spatial
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb]: From mapping document:

> 752 (separate subfields with double hyphen); else 650$z; 651$a; 662 [see defs of subfield codes at http://www.loc.gov/marc/bibliographic/bd662.html]

File: heidrun/ufl_marc.rb [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                            !df['marc:subfield'].match_attribute(:code, 'c').empty?)) ||
+                      (df.tag == '340' && 
+                       !df['marc:subfield'].match_attribute(:code, 'b').empty?) }
+            .field('marc:subfield')
+    
+    #genre 
+
+    #identifier 
+
+    #language
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb]: From mapping document:

> 041$a; OR 008 (positions 35-37)

File: heidrun/ufl_marc.rb [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            .select { |df| (df.tag == '300' && 
+                           (!df['marc:subfield'].match_attribute(:code, 'a').empty? || 
+                            !df['marc:subfield'].match_attribute(:code, 'c').empty?)) ||
+                      (df.tag == '340' && 
+                       !df['marc:subfield'].match_attribute(:code, 'b').empty?) }
+            .field('marc:subfield')
+    
+    #genre 
+
+    #identifier
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb]: From mapping document:

> 001; 020 [prefix ="ISBN: "]; 022$a [prefix ="ISSN: "]; 035$a; 050$a$b [prefix ="LC call number: "]

File: heidrun/ufl_marc.rb [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    
+    # for relation below the mapping is to both 780$t and 787$t, not sure how
+    # to combine attribute mappings in an OR relationship
+    relation record.field('marc:datafield').match_attribute(:tag, '780')
+                   .field('marc:subfield').match_attribute(:code, 't')            
+                 
+    rights record.field('marc:datafield').match_attribute(:tag, '506')
+                   .field('marc:subfield').match_attribute(:code, 'a')
+
+    #subject
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb]: From mapping document:

> 600; 61X; 650; 651; 653; 654; 655; 656; 657; 658; 69X (separate subfield with double hyphens)

File: heidrun/ufl_marc.rb [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    # to combine attribute mappings in an OR relationship
+    relation record.field('marc:datafield').match_attribute(:tag, '780')
+                   .field('marc:subfield').match_attribute(:code, 't')            
+                 
+    rights record.field('marc:datafield').match_attribute(:tag, '506')
+                   .field('marc:subfield').match_attribute(:code, 'a')
+
+    #subject 
+
+    #title
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb]: From mapping document:

> 245 (all subfields except $c); 242; 240

File: heidrun/ufl_marc.rb (hunk 3 above)
Comment (no-reply) [inline heidrun/ufl_marc.rb]: We should have `Array#join`. Or, at least, we could fairly easily.

File: heidrun/ufl_marc.rb [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+    #creator 
+
+    date :class => DPLA::MAP::TimeSpan,
+         :each => record.field('marc:datafield')
+                        .match_attribute(:tag, '260'),
+         :as => :date do
+      providedLabel date.field('marc:subfield').match_attribute(:code, 'c')
+    end
+
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb:201]: missing mapping (or comment) for `dcformat` - from mapping document:

> 007 position 00 [see http://www.loc.gov/marc/bibliographic/bd007.html]; position 06 in Leader [see “06 - Type of record“ here: http://www.loc.gov/marc/bibliographic/bdleader.html]; 337$a; 338$a; 340$a

File: heidrun/ufl_marc.rb [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+                   .field('marc:subfield').match_attribute(:code, 't')            
+                 
+    rights record.field('marc:datafield').match_attribute(:tag, '506')
+                   .field('marc:subfield').match_attribute(:code, 'a')
+
+    #subject 
+
+    #title 
+
+    #dctype
```
Comment (anarchivist) [inline heidrun/ufl_marc.rb]: this should be the same thing for now as the mapping as `dcformat`

File: heidrun/ufl_marc.rb (hunk 3 above)
Comment (no-reply) [inline heidrun/ufl_marc.rb]: I'm concerned about this mapping. The first 1000 records contain exactly one match, and it doesn't look like a description.

**Comment by markbreedlove** [comment]
I'm closing this, and will open a new one.

**Pull request closed without merging**